packages/sanity/src/desk/comments/src/context/onboarding/CommentsOnboardingProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {useCallback, useMemo, useState} from 'react'
+import {CommentsOnboardingContext} from './CommentsOnboardingContext'
+import {CommentsOnboardingContextValue} from './types'
+
+const VERSION = 1
+const LOCAL_STORAGE_KEY = `sanityStudio:comments:inspector:onboarding:dismissed:v${VERSION}`
+
+const setLocalStorage = (value: boolean) => {
+  window.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(value))
+}
+
+const getLocalStorage = (): boolean => {
+  const value = window.localStorage.getItem(LOCAL_STORAGE_KEY)
+  return value ? JSON.parse(value) : false
```

Comment:
I would recommend wrapping this in a try catch and silently ignore any parse error. If, for some reason the local storage value isn't valid stringified json this will throw a disruptive error. The last thing we want is to ask people to clear their localstorage. Localstorage can be finicky, and sensitive to various security policies in incognito/private mode, so I would also consider adding error handling around every localstorage call making sure it basically cannot throw an error.

Reply:
Thanks, good point – I have now wrapped these in try/catch 👍